chronix-server-compaction-handler/src/main/java/de/qaware/chronix/solr/compaction/LazyCompactor.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2016 QAware GmbH
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package de.qaware.chronix.solr.compaction;
+
+import de.qaware.chronix.converter.common.DoubleList;
+import de.qaware.chronix.converter.common.LongList;
+import de.qaware.chronix.timeseries.MetricTimeSeries;
+import org.apache.lucene.document.Document;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.schema.IndexSchema;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import static de.qaware.chronix.solr.compaction.ListUtils.subList;
+import static de.qaware.chronix.solr.compaction.ListUtils.sublist;
+import static java.lang.Math.min;
+
+/**
+ * Takes documents and merges them to larger ones.
+ *
+ * @author alex.christ
+ */
+public class LazyCompactor {
+    private int threshold;
+
+    /**
+     * Creates an instance.
+     *
+     * @param pointsPerChunk the number of data points to be merged into a single document.
+     */
+    public LazyCompactor(int pointsPerChunk) {
+        this.threshold = pointsPerChunk;
+    }
+
+    /**
+     * Merges documents into larger ones
+     *
+     * @param documents the documents to compact
+     * @param schema    the current solr schema
+     * @return the compaction result
+     */
+    public Iterable<CompactionResult> compact(Iterable<Document> documents, IndexSchema schema) {
+        return new LazyCompactionResultSet(documents, schema);
+    }
+
+    private class LazyCompactionResultSet implements Iterator<CompactionResult>, Iterable<CompactionResult> {
```

Comment:
\[Checkstyle\] ERROR: Class LazyCompactionResultSet should be declared as final\.